protocol/x/clob/keeper/msg_server_proposed_operations.go

```diff
@@ -18,12 +19,10 @@ func (k msgServer) ProposedOperations(
 		ctx,
 		msg.GetOperationsQueue(),
 	); err != nil {
-		panic(
-			errorsmod.Wrapf(
-				err,
-				"Block height: %d",
-				ctx.BlockHeight(),
-			),
+		return nil, errorsmod.Wrapf(
+			err,
+			"Block height: %d",
+			ctx.BlockHeight(),
```

Comment:
for my understanding - where are these errors being logged? 

previously this could be logged by the panic logging middleware. if we make this switch, do we need to update any of our alerts/monitors accordingly? if so, can you also own this?

Reply:
will look into this. Intuitively it doesn't make sense to panic here but if we have some monitoring and alerts that are reliant on this I might leave it that way for now. This will change the DeliverTx process such that instead of panicking, the DeliverTx ABCI callback returns an error and I'm admittedly not totally sure what happens in terms of logging when that happens

Reply:
It seems like out of the box errors returned by `runMsgs` during `runTx` (the function that runs transactions during DeliverTx) are not logged. Options here would be to either
1: Add a line in `runTx` in our cosmos sdk fork to call `ctx.Logger().Error(err)` when an error is returned from `runMsgs`
2: just revert this and allow the panic recovery middleware to do the logging as it has been. Most of our message handlers don't seem to panic when an error is encountered, it seems that `ProposedOperations` is an exception here. Seems like the sole reason we are panicking is to make use of the logging in the recovery middleware. I can try to find out if any of our alarms rely on these logs. My thought would be that these logs contribute to the server error logs alarm we have set up, but nothing else

Reply:
tbf though I don't think these errors should be surfacing in the error logs, as an error was encountered but it was not a critical application error, just an issue we gracefully handled. Think I'll keep this change as is

Reply:
we need some visibility into this so that we can debug issues. Proposed operations should never fail, unlike other messages like order placement, which can fail for a variety of reasons. so in a sense, it is a critical application error/logic bug.

imo we can just add the ctx.Logger().Error() here before we return. no need to modify cosmos sdk

Reply:
Why is it that proposed operations should never fail? We don't do stateful validation until we reach DeliverTx and sometimes that validation fails. Reaching an error here just means that the block proposer proposed an invalid block, not that there was an application error or logic bug. In any case, to preserve the log I can add that `ctx.Logger().Error()` in this handler

Reply:
A key difference between proposed operations vs. order placement/cancellation is that the operations are protocol injected. Since order placement/cancellation are user submitted, they can contain errors and arbitrary data that are invalid, and hence by rejected by the protocol.

On the other hand, there should be no reason for a honest proposer to inject invalid operations. Order matching should always use the latest state to make sure that we don't overfill orders, etc. This is kind of important since all proposed operations are in one transaction and is all-or-none. If proposers are constantly proposing operations that are rejected, the chain would be useless.

And yes, proposed operations can fail validation if it was proposed by a malicious proposer (dydx-research), but in this case, the operations queue has been tampered with so it's a different issue.